### ISSUE TITLE
V3 QueryObserver does not respect options set via queryClient.setQueryDefaults

### ISSUE THREAD

**Issue by tikotzky** [tannerlinsley/react-query#1219]
**Describe the bug**
Options set via `queryClient.setQueryDefaults` are not always respected.
The option I was testing with was `refetchOnWindowFocus`, though it seems to be an issue with any option passed to the `QueryObserver`.

for example, a query matching this default will still run on window focus even though `refetchOnWindowFocus` is set to false
```js
queryClient.setQueryDefaults(['todos'], { refetchOnWindowFocus: false });
```

**To Reproduce**
You can view a sandbox with a reproduction here:
https://codesandbox.io/s/loving-pasteur-hg2jz?file=/src/index.js:393-667

**Expected behavior**
The query would respect all options set via `queryClient.setQueryDefaults` and in this case should not refetch on window focus.

**Comment by TkDodo** [comment]
seems like there is a fix already, can you verify @tikotzky ?

https://github.com/tannerlinsley/react-query/releases/tag/v3.2.0-beta.13
https://github.com/tannerlinsley/react-query/commit/65f9f1283cd65270a57a45e35891d7c36afec620

**Comment by tikotzky** [comment]
@TkDodo  that fix was for issue https://github.com/tannerlinsley/react-query/issues/1214

The sandbox linked above is already using beta.13 and still has the issue.

**Comment by boschni** [comment]
Should be fixed in `3.2.0-beta.15`

**Comment by tikotzky** [comment]
Seems to be working correctly now
see: https://codesandbox.io/s/angry-bhaskara-uc24k?file=/src/index.js

Thanks!
